src/Packagist/WebBundle/Controller/WebController.php

```diff
@@ -89,8 +89,8 @@ public function searchAction(Request $req)
             if ($perPage <= 0 || $perPage > 100) {
                 $perPage = max(0, min(100, $perPage));
             }
-
-            $allowed = $this->isGranted('ROLE_MAINTAINER') ? null :
+            $allowed = $this->isGranted('ROLE_MAINTAINER') || 
+              $this->container->getParameter('packagist_web.allow_read_only_access') ? null :
                 $this->getDoctrine()
                     ->getRepository('PackagistWebBundle:Group')
                     ->getAllowedPackagesForUser($this->getUser(), false);
```

Comment:
With read only patch, we get this error:

[2022-01-20 20:32:55] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Type error: Argument 1 passed to Packagist\WebBundle\Repository\GroupRepository::ge
tAllowedPackagesForUser() must be an instance of Packagist\WebBundle\Entity\User, null given, called in /var/www/packagist/src/Packagist/WebBundle/Controller/WebController.php on line 361" at /var/www/p
ackagist/src/Packagist/WebBundle/Repository/GroupRepository.php line 49 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Type error: Argument 1 passed to Packa
gist\\WebBundle\\Repository\\GroupRepository::getAllowedPackagesForUser() must be an instance of Packagist\\WebBundle\\Entity\\User, null given, called in /var/www/packagist/src/Packagist/WebBundle/Cont
roller/WebController.php on line 361 at /var/www/packagist/src/Packagist/WebBundle/Repository/GroupRepository.php:49)